core/src/main/java/tc/oc/pgm/settings/SettingsMatchModule.java

```diff
@@ -0,0 +1,173 @@
+    int obsToolsRows = (tools.length + 9 - 1) / 9;
+
+    InventoryMenuBuilder obsBuilder =
+        new InventoryMenuBuilder(manager, SETTINGS_ROWS + obsToolsRows);
+    InventoryMenuBuilder otherBuilder = new InventoryMenuBuilder(manager, SETTINGS_ROWS);
+
+    obsBuilder.setName(TranslatableComponent.of(INVENTORY_TITLE, TextColor.AQUA));
+    otherBuilder.setName(TranslatableComponent.of(INVENTORY_TITLE, TextColor.AQUA));
+
+    for (int i = 0; i < SettingKey.values().length; i++) {
+      SettingKey key = SettingKey.values()[i];
+      InventoryItem invItem =
+          InventoryItemBuilder.createItem(
+                  manager, (menu, player) -> createSettingMenuItem(match.getPlayer(player), key))
+              .onClick(
+                  (menu, player) -> {
+                    match.getPlayer(player).getSettings().toggleValue(key);
+                    menu.invalidate(player);
+                  })
+              .build();
+
+      obsBuilder.addItem(0, i, invItem);
+      otherBuilder.addItem(0, i, invItem);
+    }
+
+    for (int i = 0; i < tools.length; i++) {
+      ObserverTool tool = tools[i];
+      obsBuilder.addItem(
+          1,
+          i,
+          InventoryItemBuilder.createItem(manager, (menu, player) -> tool.createItem(player))
+              .onClick(tool));
+    }
+
+    this.observerMenu = obsBuilder.build();
+    this.otherMenu = otherBuilder.build();
+  }
+
+  @Override
+  public void unload() {
+    manager.disable();
+  }
+
+  @EventHandler
+  public void onObserverKitApply(ObserverKitApplyEvent event) {
+    refreshKit(event.getPlayer());
+  }
+
+  @EventHandler
+  public void onToolClick(PlayerInteractEvent event) {
+    if (isRightClick(event.getAction())) {
+      ItemStack item = event.getPlayer().getItemInHand();
+
+      if (item.getType().equals(TOOL_MATERIAL)) {
+        MatchPlayer player = match.getPlayer(event.getPlayer());
+        openMenu(player);
+      }
+    }
+  }
+
+  public void openMenu(MatchPlayer player) {
+    if (player.isObserving()) {
+      observerMenu.open(player.getBukkit());
+    } else {
+      otherMenu.open(player.getBukkit());
+    }
+  }
+
+  private void refreshKit(MatchPlayer player) {
+    if (player.isObserving()) {
+      player.getInventory().setItem(TOOL_BUTTON_SLOT, createToolItem(player.getBukkit()));
+    }
+  }
+
+  private boolean isRightClick(Action action) {
+    return action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK;
+  }
+
+  private ItemStack createToolItem(Player player) {
+    Component displayName =
+        TranslatableComponent.of(INVENTORY_TITLE, TextColor.AQUA, TextDecoration.BOLD);
+    Component lore = TranslatableComponent.of("setting.lore", TextColor.GRAY);
+
+    return new ItemBuilder()
+        .material(TOOL_MATERIAL)
+        .name(TextTranslations.translateLegacy(displayName, player))
+        .lore(TextTranslations.translateLegacy(lore, player))
```

Comment:
Edit `ItemBuilder` to also accept in `Component`s. In the future, I'm thinking it can do some magic to update translations in the background.